test/framework/status.go

```diff
@@ -120,3 +121,52 @@ func (f *Framework) WaitForResourceAvailable(ctx context.Context, getResourceSta
 
 	return nil
 }
+
+// WaitForConfigResourceAcceptedCondition waits for a configuration resource (serviceMonitor, podMonitor, scrapeConfig and probes) to meet the expected Accepted condition.
+// If the condition isn't met within the given timeout, it returns an error.
+func (f *Framework) WaitForConfigResourceAcceptedCondition(ctx context.Context, getConfigResourceStatus func(context.Context) ([]monitoringv1.WorkloadBinding, error), workload metav1.Object, resource string, acceptedStatus monitoringv1.ConditionStatus, timeout time.Duration) error {
+	var pollErr error
+	if err := wait.PollUntilContextTimeout(ctx, 5*time.Second, timeout, true, func(ctx context.Context) (bool, error) {
+		var bindings []monitoringv1.WorkloadBinding
+		bindings, pollErr = getConfigResourceStatus(ctx)
+		if pollErr != nil {
+			return false, nil
+		}
+
+		var bindingFound bool
+		for _, binding := range bindings {
+			if binding.Resource == resource && binding.Name == workload.GetName() && binding.Namespace == workload.GetNamespace() {
+				bindingFound = true
+				if len(binding.Conditions) == 0 {
+					pollErr = fmt.Errorf("expected binding for resource %q with name %q in namespace %q to have conditions, but got none", resource, workload.GetName(), workload.GetNamespace())
+					return false, nil
+				}
+
+				var foundCondition bool
+				for _, cond := range binding.Conditions {
+					if cond.Status == acceptedStatus {
+						foundCondition = true
+						break
```

Comment:
If we return here, it would also simplify the non-happy code path since we don't need to check for a boolean.
```suggestion
						return true, nil